admin/billing/payment/stripe.go

```diff
@@ -54,19 +54,32 @@ func (s *Stripe) FindCustomer(ctx context.Context, customerID string) (*Customer
 		Customer: stripe.String(c.ID),
 	})
 
+	hasPaymentMethod := false
+	var isCardValid *bool
+	// very basic check if the customer has a payment method and if it's a card then is not expired
+	for i.Next() {
+		hasPaymentMethod = true
+		pm := i.PaymentMethod()
+		if pm.Type == stripe.PaymentMethodTypeCard {
+			isCardValid = new(bool)
+			*isCardValid = int(pm.Card.ExpYear) > time.Now().Year() || (int(pm.Card.ExpYear) == time.Now().Year() && int(pm.Card.ExpMonth) >= int(time.Now().Month()))
+		}
+	}
```

Comment:
Does Stripe have an API for this? There are other factors than just expiry, and Stripe might know about them (like probably they track cards they have detected to be blocked)

Reply:
I could not find any such API but added specifically this check because Eric O mentioned that we want to show a banner if the card expires.